src/Controls/src/Core/Button/Button.Android.cs

```diff
@@ -1,13 +1,35 @@
 ﻿#nullable disable
-using System;
-using System.Collections.Generic;
-using System.Text;
+using Google.Android.Material.Button;
 using Microsoft.Maui.Controls.Platform;
 
 namespace Microsoft.Maui.Controls
 {
 	public partial class Button
 	{
+		MaterialButton _materialButton;
+
+		private protected override void OnHandlerChangedCore()
+		{
+			base.OnHandlerChangedCore();
+
+			if (Handler != null)
+			{
+				if (Handler is ButtonHandler buttonHandler && buttonHandler.PlatformView is MaterialButton materialButton)
+				{
+					_materialButton = materialButton;
+					_materialButton.LayoutChange += OnButtonLayoutChange;
```

Comment:
This is going to cause memory leaks. 

Can we trigger this logic from the handler without events?

Reply:
Reviewed and added a way to always unsubscribe with test to validate that the Button does not leak.
<img width="394" alt="Captura de pantalla 2023-07-17 a las 11 34 18" src="https://github.com/dotnet/maui/assets/6755973/0f82229f-344f-4c8c-b32f-aacd58f24e34">